tests/test_workspace.py

```diff
@@ -25,334 +34,510 @@
+    workspace = Workspace(Resolver(), directory=str(tmp_path))
+
+    # before act
+    # TODO clear semantics
+    # requires rather odd additional path-setting because root path from
+    # workspace is not propagated - works only if called inside workspace
+    # which can be achieved with pushd_popd functionalities
+    ocrd_file = next(workspace.mets.find_files(ID='OCR-D-GT-SEG-WORD_0001'))
+    relative_name = ocrd_file.local_filename
+    ocrd_file.local_filename = join(tmp_path, relative_name)
+    pcgts_before = page_from_file(ocrd_file)
+    # before assert
+    assert pcgts_before.get_Page().imageFilename == 'OCR-D-IMG/OCR-D-IMG_0001.tif'
+
+    # act
+    workspace.rename_file_group('OCR-D-IMG', 'FOOBAR')
+    next_ocrd_file = next(workspace.mets.find_files(ID='OCR-D-GT-SEG-WORD_0001'))
+    next_ocrd_file.local_filename = join(tmp_path, relative_name)
+    pcgts_after = page_from_file(next_ocrd_file)
+
+    # assert
+    assert pcgts_after.get_Page().imageFilename == 'FOOBAR/FOOBAR_0001.tif'
+    assert Path(tmp_path / 'FOOBAR/FOOBAR_0001.tif').exists()
+    assert not Path('OCR-D-IMG/OCR-D-IMG_0001.tif').exists()
+    assert workspace.mets.get_physical_pages(for_fileIds=['OCR-D-IMG_0001']) == [None]
+    assert workspace.mets.get_physical_pages(for_fileIds=['FOOBAR_0001']) == ['phys_0001']
+
+
+def test_remove_file_group_invalid_raises_exception(sbb_data_workspace):
+    with pytest.raises(Exception) as no_fg_exc:
+        # should fail
+        sbb_data_workspace.remove_file_group('I DO NOT EXIST')
+    assert "No such fileGrp" in str(no_fg_exc.value)
+
+
+def test_remove_file_group_force(sbb_data_workspace):
+
+    # TODO
+    # check function and tests semantics
+    # should succeed
+    assert not sbb_data_workspace.remove_file_group('I DO NOT EXIST', force=True)
+    # should also succeed
+    sbb_data_workspace.overwrite_mode = True
+    assert not sbb_data_workspace.remove_file_group('I DO NOT EXIST', force=False)
+
+
+def test_remove_file_group_rmdir(sbb_data_tmp, sbb_data_workspace):
+    assert exists(join(sbb_data_tmp, 'OCR-D-IMG'))
+    sbb_data_workspace.remove_file_group('OCR-D-IMG', recursive=True)
+    assert not exists(join(sbb_data_tmp, 'OCR-D-IMG'))
+
+
+def test_remove_file_group_flat(plain_workspace):
+    """
+    https://github.com/OCR-D/core/issues/728
+    """
+
+    # act
+    added_res = plain_workspace.add_file('FOO', ID='foo', mimetype='foo/bar', local_filename='file.ext', content='foo', pageId=None).url
+    # requires additional prepending of current path because not pushd_popd-magic at work
+    added_path = Path(join(plain_workspace.directory, added_res))
+
+    # assert
+    assert added_path.exists()
+    plain_workspace.remove_file_group('FOO', recursive=True)
+
+
+@pytest.fixture(name='kant_complex_workspace')
+def _fixture_kant_complex(tmp_path):
+    shutil.copytree(assets.path_to('kant_aufklaerung_1784-complex/data'), str(tmp_path), dirs_exist_ok=True)
```

Comment:
`dirs_exist_ok` was introduced in Python 3.8, but we need to support 3.7 for a while longer.

Reply:
And we should phase out 3.6 as soon as possible, that has reached end of life now.

Reply:
I added a workaround in ced76d05